### DISCUSSION TITLE
How to send piece of HTML from Webview to backend

### DISCUSSION THREAD

**Discussion by dougkirkley** [wailsapp/wails#2619] [Q&A]
Good morning,

I am primarily a backend and CLI focused developer. I currently have a tool that allows users to retrieve AWS credentials via Assume role with SAML. 

To do this currently involves opening a Chromedp browser to send users to the endpoint in AWS for our Google identity provider. This requires a user to be logged in via Google, then it will send the SAML response to AWS to log in. Currently Chromedp can't be run in background because Google makes you sign again due to bot checking. 

What is the best way to send a user to an external website in Webview that can send the part of the document with the SAML response in it to the backend? I know I can send events from the frontend to the backend, but since I only need to navigate to a website I'm not sure if it makes sense to just wrap an href redirect inside JS vs specifying a URL in the Webview window and executing some JS code to send the data back.

Thank you for your time

**Comment by FranPregernik** [comment]
I have a use-case for this as well.

I render data on the UI with Apache ECharts and it does a great job. 
I would like to get that image and put it in a PDF that I generate in GO.
There is no go lib that has all the features that I need.

This go [lib](https://github.com/go-echarts/go-echarts)  comes really close to supporting the features  of echarts and has this [render](https://go-echarts.github.io/go-echarts/#/en-us/render-to-image) functionality which uses the chrome debug protocol to render it there get back an image back to go.

I would love it if I could attach it to the webkit process that wails runs and use the "same" JS to render the UI and get an image to include in PDF.
